go/tools/builders/cover_test.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+type test struct {
+	name string
+	in   string
+	out  string
+}
+
+var tests = []test{
+	{
+		name: "no imports",
+		in: `package main
+`,
+		out: `package main; import "github.com/bazelbuild/rules_go/go/tools/coverdata"
+
+func init() {
+	coverdata.RegisterFile("srcName",
+		varName.Count[:],
+		varName.Pos[:],
+		varName.NumStmt[:])
+}
+`,
+	},
+	{
+		name: "other imports",
+		in: `package main
+
+import (
+	"os"
+)
+`,
+		out: `package main; import "github.com/bazelbuild/rules_go/go/tools/coverdata"
+
+import (
+	"os"
+)
+
+func init() {
+	coverdata.RegisterFile("srcName",
+		varName.Count[:],
+		varName.Pos[:],
+		varName.NumStmt[:])
+}
+`,
+	},
+	{
+		name: "existing import",
+		in: `package main
+
+import "github.com/bazelbuild/rules_go/go/tools/coverdata"
+`,
+		out: `package main
+
+import "github.com/bazelbuild/rules_go/go/tools/coverdata"
+
+func init() {
+	coverdata.RegisterFile("srcName",
+		varName.Count[:],
+		varName.Pos[:],
+		varName.NumStmt[:])
+}
+`,
+	},
+	{
+		name: "existing _ import",
+		in: `package main
+
+import _ "github.com/bazelbuild/rules_go/go/tools/coverdata"
+`,
+		out: `package main
+
+import coverdata "github.com/bazelbuild/rules_go/go/tools/coverdata"
+
+func init() {
+	coverdata.RegisterFile("srcName",
+		varName.Count[:],
+		varName.Pos[:],
+		varName.NumStmt[:])
+}
+`,
+	},
+	{
+		name: "existing renamed import",
+		in: `package main
+
+import cover0 "github.com/bazelbuild/rules_go/go/tools/coverdata"
+`,
+		out: `package main
+
+import cover0 "github.com/bazelbuild/rules_go/go/tools/coverdata"
+
+func init() {
+	cover0.RegisterFile("srcName",
+		varName.Count[:],
+		varName.Pos[:],
+		varName.NumStmt[:])
+}
+`,
+	},
+}
+
+func TestRegisterCoverage(t *testing.T) {
+	const filename = "test_input.go"
```

Comment:
```suggestion
	var filename = filepath.Join(t.TempDir(), "test_input.go")
```
The current directory of the test may not be writable, or may happen to already have a file called `test_input.go`. It's better to write to a temp dir. By using `t.TempDir()`, you don't need to do `defer os.Remove(filename)` either.

Reply:
Ah, I forgot about `testing.(*T).TempDir`, thanks. Updated